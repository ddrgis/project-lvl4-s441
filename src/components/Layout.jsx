import React from 'react';
import AlertModal from './AlertModal';

export default function Layout({ component: Component }) {
  return (
    <>
      <header>
        <nav className="header navbar navbar-expand-lg navbar-light bg-faded border-bottom">
          <h1 className="navbar-brand mb-0">React Chat</h1>
          <button
            type="button"
            className="navbar-toggler"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
          >
            <span className="navbar-toggler-icon" />
          </button>
        </nav>
      </header>
      <main className="main container-fluid">
        <Component />
      </main>
      <AlertModal />
    </>
  );
}
