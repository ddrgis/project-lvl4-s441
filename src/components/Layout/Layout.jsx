import React from 'react';
import { Route } from 'react-router';
import { NavLink } from 'react-router-dom';
import ChannelDeletionModal from '../ChannelDeletionModal/ChannelDeletionModal';
import AlertModal from '../AlertModal/AlertModal';

export default function Layout({ component: Component, route, ...props }) {
  return (
    <>
      <nav className="header navbar navbar-expand-lg navbar-light bg-faded border-bottom">
        <NavLink className="navbar-brand" to="/">
          {'Chat'}
        </NavLink>
        <button
          type="button"
          className="navbar-toggler"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse">
          <ul className="navbar-nav mr-auto">
            <li className="nav-item active">
              <NavLink className="nav-link" to="/">
                {'Home'}
              </NavLink>
            </li>
          </ul>
        </div>
      </nav>
      <div className="main container-fluid">
        <Route route={route} render={() => <Component {...props} />} />
      </div>
      <ChannelDeletionModal />
      <AlertModal />
      <footer className="footer align-items-end">
        <div className="container-fluid">Footer</div>
      </footer>
    </>
  );
}
