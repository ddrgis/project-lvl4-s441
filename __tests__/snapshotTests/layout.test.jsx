import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Layout from '../../src/components/Layout';

jest.mock('../../src/components/AlertModal/AlertModal', () => () => <div />);

it('renders correctly', () => {
  const mockComponent = () => <></>;
  const tree = renderer
    .create(
      <MemoryRouter>
        <Layout path="/" component={mockComponent} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
