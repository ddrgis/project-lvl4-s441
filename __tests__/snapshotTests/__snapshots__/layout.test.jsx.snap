// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="header navbar navbar-expand-lg navbar-light bg-faded border-bottom"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Chat
    </a>
    <button
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="main container-fluid"
  />,
  <div />,
  <footer
    className="footer align-items-end"
  >
    <div
      className="container-fluid"
    >
      Footer
    </div>
  </footer>,
]
`;
